### ISSUE TITLE
Files not exported to defined file location

### ISSUE THREAD

**Issue by yuseferi** [drupal-tome/tome-project#13]
I have the following code

```
  /**
   * @param $uri
   */
  public function generateStaticOfUri($uri) {
    $path = $uri;
    TomeStaticHelper::setBaseUrl(\Drupal::request(), self::LOCAL_STORE_HOST);
    $this->staticGenerator->getStaticDirectory();
    $this->requestPreparer->prepareForRequest();
    $invoke_paths = $this->staticGenerator->requestPath($path);
    $invoke_paths = $this->staticGenerator->exportPaths($invoke_paths);
    foreach ($invoke_paths as $path) {
      if (pathinfo($path, PATHINFO_EXTENSION)) {
        $this->requestPreparer->prepareForRequest();
        $this->staticGenerator->requestPath($path);
      }
    }
  }

```

and I added this config in  `settings.php`

```
$settings['tome_static_directory'] = '../static/html';
$settings['tome_files_directory'] = '../static/files';

```

but when I run my code, the files ( images, assets downloaded to drupal file path `static/html/sites/default/files` path.

I expected the downloaded to `../static/files` and the src and link of assets changes to the new location.

![image](https://user-images.githubusercontent.com/1736832/155364393-0f2ee279-ca5d-462b-abcc-6e02c6f3937a.png)

did I do something wrong?

how can I  do this ?

**Comment by mortenson** [comment]
`tome_files_directory` is for the original files exported by Tome Sync, not static HTML from Tome Static.

**Comment by yuseferi** [comment]
isn't any way to make it doable?

**Comment by mortenson** [comment]
This is more of a general Drupal issue than a Tome one - could you use`$settings['file_public_path']` ?

**Comment by yuseferi** [comment]
@mortenson 
thanks for replying.
I tried, it's just changing Drupal public. path.
A solution which comes to my mind is, 
before the HTML is exported,  get the HTML of the pages, go through all the assets save them in `tome_files_directory` and change the origin file path in the origin HTML and save it.

I can do this stuffs with Raw PHP codes, but I'm wondering, is there any function or solution to do it with TOME?

Thanks

**Comment by mortenson** [comment]
@yuseferi No, no plans right now. There is `\Drupal\tome_static\Event\TomeStaticEvents::FILE_SAVED` and `\Drupal\tome_static\Event\TomeStaticEvents::MODIFY_HTML`, but I think a bash script might be easier. Might be able to do it in two lines - first create a symlink (or move) at `static/html/files -> static/html/sites/default/files`, then do a sed in place recursively to replace `sites/default/files/` with `files/`.

**Comment by yuseferi** [comment]
@mortenson bash scripts looks interesting.
I want to explain why I want it.

we have a Drupal website with several books. ( created by Drupal book module, it could contain several nodes).
we want to develop an application that lets users download books individually and use them offline.
over the night periodically I want to generate static versions of books, but them in a structure in which users can add another book to  his/her offline book list easily.

I want to minimize the size of each book, I don't want if a user wants to download a book, I add other books assets( mostly photos), to what he/she download. and at the same when users wants to have more than one book offline , create several local servers in the mobile app to serve them.


so my ideal solutions is  
```
/
/book1/
/book1/chapter1/
/book1/chapter1/.....
/book2/ 
/files
// general and theme files 
```
I hope I could explain what I want.
